docs/addons/argo-cd.md

```diff
@@ -25,9 +27,18 @@ new EksBlueprint(app, 'my-stack-name', addOns, [], {
 });
 ```
 
-The above will use `argocd` namespace and install all the components. In order to bootstrap workloads you will need to change the default ArgoCD admin password and add repositories as specified in the [Getting Started](https://argoproj.github.io/argo-cd/getting_started/#port-forwarding) documentation.
+The above will create an `argocd` namespace and install all Argo CD components. In order to bootstrap workloads you will need to change the default ArgoCD admin password and add repositories as specified in the [Getting Started](https://argoproj.github.io/argo-cd/getting_started/#port-forwarding) documentation.
+
+## Functionality
+
+1. Creates the namespace specified in the construction parameter (`argocd` by default).
+2. Deploys the [`argo-cd`](https://argoproj.github.io/argo-helm) Helm chart into the cluster.
+3. Allows to specify `ApplicationRepository` selecting the required authentication method as SSH Key, username/password or username/token. Credentials are expected to be set in AWS Secrets Manager and replicated to the desired region. If bootstrap repository is specified, creates the initial bootstrap application which may be leveraged to bootstrap workloads and/or other add-ons through GitOps.
+4. Allows setting the initial admin password through AWS Secrets Manager, replicating to the desired region. 
+
+## Setting an Admin Password
 
-## Setting Admin Password from AWS Secrets Manager
+By default, the Argo CD add-on will create a new admin password for you. To specify you own, you can leverage the AWS Secrets manage integration.
```

Comment:
minor: you own = your own

Reply:
minor: leverage the AWS Secrets manage integrations = (maybe) leverage the AWS Secrets Manager to provide the secret.